File: ch28/Chapter28-VB/App_Code/Listing28-52.vb
Imports Microsoft.VisualBasic

Public Class MyCompanySettings
    Inherits ConfigurationSection

    <ConfigurationProperty("Key1", DefaultValue:="This is the value of Key 1", _
       IsRequired:=False)> _
    Public ReadOnly Property Key1() As String
        Get
            Return MyBase.Item("Key1").ToString()
        End Get
    End Property

    <ConfigurationProperty("Key2", IsRequired:=True)> _
    Public ReadOnly Property Key2() As String
        Get
            Return MyBase.Item("Key2").ToString()
        End Get
    End Property
End Class

File: ch28/Chapter28-CS/App_Code/Listing28-52.cs
using System.Configuration;
        
public class MyCompanySettings : ConfigurationSection
{
    [ConfigurationProperty("Key1", DefaultValue = "This is the value of Key 1",
     IsRequired = false)]
    public string Key1
    {
        get
        {
            return this["Key1"] as string;
        }
    }
        
    [ConfigurationProperty("Key2", IsRequired = true)]
    public string Key2
    {
        get
        {
            return this["Key2"] as string;
        }
    }
}